From 50ed977e1b166ba23a029f3a5194121ea4d45dd7 Mon Sep 17 00:00:00 2001
From: Thomas Perl <m@thp.io>
Date: Fri, 6 Feb 2015 16:28:43 +0100
Subject: [PATCH] zypp/PublicKey.cc: Use GPG_BINARY from KeyRing.cc

On systems where "gpg" is not available, checking the public key
would fail. Use GPG_BINARY in both PublicKey.cc and KeyRing.cc,
and define it in a single place only (ExternalProgram.h).

The original patch was partially integrated in upstream 
commit 092cded378742fcbb0761cc470da863cb32ed537, but only as a duplicated
define in KeyRing and PublicKey. This patch now just moves the define to 
ExternalProgram.h
---
 zypp/ExternalProgram.h | 2 ++
 zypp/KeyRing.cc        | 3 ---
 zypp/PublicKey.cc      | 3 ---
 3 files changed, 2 insertions(+), 6 deletions(-)

diff --git a/zypp/ExternalProgram.h b/zypp/ExternalProgram.h
index f3fc850..99a3143 100644
--- a/zypp/ExternalProgram.h
+++ b/zypp/ExternalProgram.h
@@ -22,6 +22,8 @@
 #include "zypp/base/ExternalDataSource.h"
 #include "zypp/Pathname.h"
 
+#define GPG_BINARY "/usr/bin/gpg2"
+
 namespace zypp {
 
     /**
diff --git a/zypp/KeyRing.cc b/zypp/KeyRing.cc
index d6b8246..826ce4c 100644
--- a/zypp/KeyRing.cc
+++ b/zypp/KeyRing.cc
@@ -35,9 +35,6 @@ using std::endl;
 #undef  ZYPP_BASE_LOGGER_LOGGROUP
 #define ZYPP_BASE_LOGGER_LOGGROUP "zypp::KeyRing"
 
-/** \todo Fix duplicate define in PublicKey/KeyRing */
-#define GPG_BINARY "/usr/bin/gpg2"
-
 ///////////////////////////////////////////////////////////////////
 namespace zypp
 { /////////////////////////////////////////////////////////////////
diff --git a/zypp/PublicKey.cc b/zypp/PublicKey.cc
index fa17b79..fdbc95e 100644
--- a/zypp/PublicKey.cc
+++ b/zypp/PublicKey.cc
@@ -28,9 +28,6 @@
 
 #include <ctime>
 
-/** \todo Fix duplicate define in PublicKey/KeyRing */
-#define GPG_BINARY "/usr/bin/gpg2"
-
 using std::endl;
 
 ///////////////////////////////////////////////////////////////////
-- 
1.8.3-rc3


